spec/classes/foreman_proxy__plugin__pulp_spec.rb

```diff
@@ -2,8 +2,16 @@
 
 describe 'foreman_proxy::plugin::pulp' do
 
+  let :puppet_environmentpath do
+    if Gem::Version.new(Puppet.version) >= Gem::Version.new('4.0')
+      {:puppet_environmentpath => '/etc/puppetlabs/code/environments'}
```

Comment:
It's merged with facts, and it's a hash here so the later spec test returns nil and the || trick works